packages/sanity/src/presentation/usePreviewUrl.ts

```diff
@@ -53,7 +53,9 @@ export function usePreviewUrl(
     const resolvedUrl = await resolvePreviewUrl({
       client,
       previewUrlSecret: previewUrlSecret!,
-      studioPreviewPerspective,
+      studioPreviewPerspective: Array.isArray(studioPreviewPerspective)
```

Comment:
Feels like we need to check if the perspective is an array somewhat often, I'm not sure if we have a helper for this in the releases package but it might be worth creating a helper function?